scripts/o3de/o3de/export_project.py

```diff
@@ -136,3 +283,402 @@ def add_args(subparsers) -> None:
+    if required and not asset_bundler_batch_path.is_file():
+        raise ExportProjectError(f"Missing the 'AssetBundlerBatch' tool, expected at '{asset_bundler_batch_path}'")
+    return asset_bundler_batch_path
+
+
+def build_assets(ctx: O3DEScriptExportContext,
+                 tools_build_path: pathlib.Path,
+                 fail_on_ap_errors: bool,
+                 logger: logging.Logger = None) -> None:
+    """
+    Build the assets for the project
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to locate AssetProcessorBatch
+    @param fail_on_ap_errors:   Option to fail the whole process if an error occurs during asset processing
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Make sure `AssetProcessorBatch` is available
+    asset_processor_batch_path = get_asset_processor_batch_path(tools_build_path, required=True)
+    if not asset_processor_batch_path.exists():
+        raise ExportProjectError("Missing AssetProcessorBatch. The pre-requisite tools must be built first.")
+
+    # Build the project assets with the {project_name}.Assets custom target
+    if logger:
+        logger.info(f"Processing assets for {ctx.project_name}")
+
+    cmake_build_assets_command = [asset_processor_batch_path, "--project-path", ctx.project_path]
+    ret = process_command(cmake_build_assets_command,
+                          cwd=ctx.engine_path if ctx.is_engine_centric else ctx.project_path)
+    if ret != 0:
+        if fail_on_ap_errors:
+            raise ExportProjectError(f"Error building assets for project {ctx.project_name}.")
+        else:
+            if logger:
+                logger.warning("Some assets failed to processed.")
+    if logger:
+        logger.info(f"Completed processing assets for {ctx.project_name}!")
+
+
+def build_export_toolchain(ctx: O3DEScriptExportContext,
+                           tools_build_path: pathlib.Path,
+                           logger: logging.Logger = None) -> None:
+    """
+    Build (or rebuild) the export tool chain (AssetProcessorBatch and AssetBundlerBatch)
+
+    @param ctx:                 Export Context
+    @param tools_build_path:    The tools (cmake) build path to create the build project for the tools
+    @param logger:              Optional Logger
+    @return: None
+    """
+
+    # Generate the project for export toolchain
+    cmake_configure_command = ["cmake", "-B", tools_build_path]
+
+    if ctx.is_engine_centric:
+        cmake_configure_command.extend(["-S", ctx.engine_path])
+    else:
+        cmake_configure_command.extend(["-S", ctx.project_path])
+
+    # Generate the project for the pre-requisite tools
+    if GENERATOR:
+        cmake_configure_command.extend(["-G", GENERATOR])
+    if CMAKE_GENERATOR_OPTIONS:
+        cmake_configure_command.extend(CMAKE_GENERATOR_OPTIONS)
```

Comment:
We don't have an argparse option for specifying an additional cmake configure argument (such as a `--preset` or any custom options the user want to configure with)

Reply:
I don't want to introduce more complexity to the arguments at this time. I only added the ability to pass through the cmake build commands because it affects performance of the build.

Reply:
I just realized that I may need to handle this situation. For simulation build workflows, PhysX 5 is needed, but that is an opt-in CMAKE configure parameter that needs to be set.